### PR TITLE
[Cosmos] Async query fixes and COMMON_OPTIONS refining

### PR DESCRIPTION
**Pull request by simorenoh**
Based on https://github.com/Azure/azure-sdk-for-python/issues/25104 - fixing this issue by removing several options from the COMMON_OPTIONS dictionary we have.

COMMON_OPTIONS is good in theory, but since our methods use different options and each have their own logic for taking what they need from the keyword arguments, we might consider making options for different _types_ of methods and applying those where they belong in the future. For now, I'm trimming the common options to include only those that are not taken care of by individual methods, and that would not affect behavior if left there. Partial work for https://github.com/Azure/azure-sdk-for-python/issues/23279.

Also added new http constants for the logging policy and split up the way some of our query tests were made, since that reduced the readability on them.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/azure/cosmos/container.py [hunk 1]
```diff
@@ -299,7 +299,7 @@ def query_items_change_feed(
         response_hook = kwargs.pop('response_hook', None)
         if partition_key_range_id is not None:
             feed_options["partitionKeyRangeId"] = partition_key_range_id
-        partition_key = kwargs.pop("partitionKey", None)
+        partition_key = kwargs.pop("partitionKey", kwargs.pop("partition_key", None))
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/azure/cosmos/container.py:302]: Wow - this seems like it was a pretty bad bug - good catch!

**Review by annatisch — COMMENTED** [review]
From what I can see I think these changes look okay...
I can't think of any combination of parameters that would be broken by this - if I was already exclusively using `request_options` or `feed_options` dicts (not sure why we ever supported this), then that should be uneffected. And the feed_options will still ultimately be overridden by any explicit args/kwargs being passed in, so from what I can see behaviour should be unchanged....
Anything I'm missing here?

**Comment by simorenoh** [comment]
@annatisch I believe your understanding is correct on all accounts 👍 I also made sure to run the tests pipeline every time I made a change to ensure we're not breaking anything just in case, but as far as I see these changes should be good.

**Review by annatisch — APPROVED** [review]

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**